x-pack/plugins/security_solution/public/management/pages/trusted_apps/view/components/create_trusted_app_form.tsx

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { ChangeEventHandler, memo, useCallback, useEffect, useMemo, useState } from 'react';
+import {
+  EuiForm,
+  EuiFormRow,
+  EuiFieldText,
+  EuiSuperSelect,
+  EuiSuperSelectOption,
+  EuiTextArea,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { EuiFormProps } from '@elastic/eui/src/components/form/form';
+import { TRUSTED_APPS_SUPPORTED_OS_TYPES } from '../../../../../../common/endpoint/constants';
+import { LogicalConditionBuilder } from './logical_condition';
+import {
+  MacosLinuxConditionEntry,
+  NewTrustedApp,
+  TrustedApp,
+} from '../../../../../../common/endpoint/types';
+import { LogicalConditionBuilderProps } from './logical_condition/logical_condition_builder';
+import { OS_TITLES } from '../constants';
+import {
+  isTrustedAppSupportedOs,
+  isWindowsTrustedApp,
+  isWindowsTrustedAppCondition,
+} from '../../state/type_guards';
+
+const generateNewEntry = (): NewTrustedApp['entries'][0] => {
+  return {
+    field: 'process.hash.*',
+    operator: 'included',
+    type: 'match',
+    value: '',
+  };
+};
+
+interface ValidationResult {
+  isValid: boolean;
+  errors: Partial<{ [k in keyof NewTrustedApp]: string }>;
+}
+const validateFormValues = (values: NewTrustedApp): ValidationResult => {
```

Comment:
we've talked about it a couple times in meetings, another validation we could do is field name uniqueness, for instance, you can only have one `hash` field, etc.  Maybe this could solved when we call `generateNewEntry()` - enforce there that fields must be unique, generate the next "available field" and when none left, disable the "AND" button.

Not required for this PR, was just putting ideas out there for more validations.

FYI @charlie-pichette @efreeti

Reply:
I opened a separate issue to track UI/UX around validation feedback. This method here is bound to change based on that feedback